flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/FileSystemJobResultStoreFileOperationsTest.java

```diff
@@ -116,46 +123,80 @@ public void testBaseDirectoryCreationOnResultStoreInitialization() throws Except
+                fileSystemJobResultStore.markResultAsCleanAsync(DUMMY_JOB_RESULT_ENTRY.getJobId());
+        assertThat(getCleanResultIdsFromFileSystem())
+                .doesNotContain(DUMMY_JOB_RESULT_ENTRY.getJobId());
+        manuallyTriggeredExecutor.triggerAll();
+        FlinkAssertions.assertThatFuture(markCleanAsync).eventuallySucceeds();
         assertThat(getCleanResultIdsFromFileSystem())
                 .containsExactlyInAnyOrder(DUMMY_JOB_RESULT_ENTRY.getJobId());
     }
 
     @Test
-    public void testStoreCleanJobResultDeletesDirtyFile() throws Exception {
-        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+    public void testStoreCleanJobResultDeletesDirtyFile() {
+        CompletableFuture<Void> dirtyResultAsync =
+                fileSystemJobResultStore.createDirtyResultAsync(DUMMY_JOB_RESULT_ENTRY);
+        assertThat(expectedDirtyFile(DUMMY_JOB_RESULT_ENTRY)).doesNotExist();
+        manuallyTriggeredExecutor.triggerAll();
+        FlinkAssertions.assertThatFuture(dirtyResultAsync).eventuallySucceeds();
         assertThat(expectedDirtyFile(DUMMY_JOB_RESULT_ENTRY)).exists().isFile().isNotEmpty();
 
-        fileSystemJobResultStore.markResultAsClean(DUMMY_JOB_RESULT_ENTRY.getJobId());
+        CompletableFuture<Void> markResultAsCleanAsync =
+                fileSystemJobResultStore.markResultAsCleanAsync(DUMMY_JOB_RESULT_ENTRY.getJobId());
+        manuallyTriggeredExecutor.triggerAll();
+        FlinkAssertions.assertThatFuture(markResultAsCleanAsync).eventuallySucceeds();
         assertThat(expectedDirtyFile(DUMMY_JOB_RESULT_ENTRY)).doesNotExist();
     }
 
     @Test
     public void testCleanDirtyJobResultTwiceIsIdempotent() throws IOException {
-        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
-        fileSystemJobResultStore.markResultAsClean(DUMMY_JOB_RESULT_ENTRY.getJobId());
-
+        CompletableFuture<Void> dirtyResultAsync =
+                fileSystemJobResultStore.createDirtyResultAsync(DUMMY_JOB_RESULT_ENTRY);
+        manuallyTriggeredExecutor.triggerAll();
+        FlinkAssertions.assertThatFuture(dirtyResultAsync).eventuallySucceeds();
+        CompletableFuture<Void> cleanResultAsync =
+                fileSystemJobResultStore.markResultAsCleanAsync(DUMMY_JOB_RESULT_ENTRY.getJobId());
+        manuallyTriggeredExecutor.triggerAll();
+        FlinkAssertions.assertThatFuture(cleanResultAsync).eventuallySucceeds();
         final byte[] cleanFileData =
                 FileUtils.readAllBytes(expectedCleanFile(DUMMY_JOB_RESULT_ENTRY).toPath());
 
-        fileSystemJobResultStore.markResultAsClean(DUMMY_JOB_RESULT_ENTRY.getJobId());
+        CompletableFuture<Void> markResultAsCleanAsync =
+                fileSystemJobResultStore.markResultAsCleanAsync(DUMMY_JOB_RESULT_ENTRY.getJobId());
+        assertThat(expectedCleanFile(DUMMY_JOB_RESULT_ENTRY))
+                .doesNotHaveSameHashCodeAs(cleanFileData);
```

Comment:
I don't get this assert? Why do we check this here? Essentially, what we want to check is that the modification date of the file is the same after each `markResultAsCleanAsync` call. :thinking:

Reply:
I've added a fixup to remove the assert.